### ISSUE TITLE
Add functions to compute Structural Hamming Distance and Structural Intervention Distance

### ISSUE THREAD

**Issue by ankurankan** [pgmpy/pgmpy#1766]
Structural Intervention Distance: https://arxiv.org/pdf/1306.1043

**Comment by jihyeseo** [comment]
I see that @Nimish-4 already implemented SHD. I am happy to try SID unless you are on it.

**Comment by Nimish-4** [comment]
@jihyeseo In case you were asking me, I am not working on SID. @ankurankan may clarify if someone else is working on it.

**Comment by ankurankan** [comment]
@Nimish-4 @jihyeseo Nobody is currently working on SID, please feel free to implement it.

**Comment by jihyeseo** [comment]
Thanks for the info. I will start on it. Feel free to assign me to this.

**Comment by aniket2405** [comment]
I'd like to work on this.

**Comment by ankurankan** [comment]
@aniket2405 Sure. @jihyeseo had already gotten quite far in implementing this (#1927). You can branch off her branch, and 1. Update the code to some of the changes in the package (for example, BayesianNetwork is now called DiscreteBayesianNetwork), 2. See if you can simplify the implementation as we were discussing in the PR.

**Comment by kchirag2023** [comment]
@ankurankan Do anyone implemented SID? if not please ,let me know i will work on it

**Comment by ankurankan** [comment]
@kchirag2023 No, the PR has a partial implementation and is still open. Would suggest continuing from that.

**Comment by rpp5501** [comment]
Hi , I'm Rishabh a student in the CodePath AI301 program. I'm interested in picking up the Structural Intervention Distance (SID) implementation from PR #1927 to get it across the finish line. I plan to branch off the existing work, update the deprecated BayesianNetwork classes to DiscreteBayesianNetwork, and finalize the adjustment set logic. Let me know if anyone is currently working on this, otherwise I will open a fresh PR linking back to this one soon!

**Comment by ankurankan** [comment]
@rpp5501 Sorry for the delay. The PR for this has stalled. Assigning it to you.

**Comment by rpp5501** [comment]
@ankurankan could you please take a look at my pull request and let me know if you know if you have any feedback.

**Comment by rpp5501** [comment]
@ankurankan
